ilastik/experimental/api/_pipelines.py

```diff
@@ -0,0 +1,256 @@
+            )
+
+    def get_probabilities(self, raw_data):
+        return self._process(raw_data, output_slot=self._predict_op.PMaps)
+
+    def _process(self, raw_data, output_slot):
+        raw_data = convert_to_vigra(raw_data)
+        input_axistags = "".join(ax.key for ax in raw_data.axistags)
+        self._check_data(raw_data)
+        self._reorder_op.Input.setValue(raw_data)
+        processed_data = output_slot.value[...]
+        return _reorder_output(processed_data, "".join(output_slot.meta.axistags.keys()), input_axistags)
+
+
+class _ObjectClassificationPipelineBase:
+    def __init__(self, project: parser.ObjectClassificationProjectBase):
+        super().__init__()
+        self._graph = Graph()
+
+        classifer = project.classifier
+        selected_object_features = project.selected_object_features
+        self._num_channels = project.data_info.num_channels
+        self._axis_order = project.data_info.axis_order
+        self._num_spatial_dims = len(project.data_info.spatial_axes)
+        self._obj_extraction_op = OpObjectExtraction(graph=self._graph)
+        self._obj_extraction_op.Features.setValue(selected_object_features.selected_features)
+        self._predict_op = OpObjectPredict(graph=self._graph)
+        self._predict_op.Classifier.setValue(classifer.instance)
+        self._predict_op.Features.connect(self._obj_extraction_op.RegionFeatures)
+        self._predict_op.SelectedFeatures.setValue(selected_object_features.selected_features)
+        self._predict_op.LabelsCount.setValue(classifer.label_count)
+
+        self._relabel_to_obj_probs = OpMultiRelabelSegmentation(graph=self._graph)
+        self._relabel_to_obj_probs.Image.connect(self._obj_extraction_op.LabelImage)
+        self._relabel_to_obj_probs.ObjectMaps.connect(self._predict_op.ProbabilityChannels)
+        self._relabel_to_obj_probs.Features.connect(self._obj_extraction_op.RegionFeatures)
+
+        self._stack_prob_channels = OpMultiArrayStacker(graph=self._graph)
+        self._stack_prob_channels.Images.connect(self._relabel_to_obj_probs.Output)
+        self._stack_prob_channels.AxisFlag.setValue("c")
+
+    def _check_data(self, data):
+        num_channels_in_data = data.channels
+        if num_channels_in_data != self._num_channels:
+            raise ValueError(
+                f"Number of channels mismatch. Classifier trained for {self._num_channels} but input has {num_channels_in_data}"
+            )
+
+        num_spatial_in_data = sum(a.isSpatial() for a in data.axistags)
+        if num_spatial_in_data != self._num_spatial_dims:
+            raise ValueError(
+                "Number of spatial dims doesn't match. "
+                f"Classifier trained for {self._num_spatial_dims} but input has {num_spatial_in_data}"
+            )
+
+
+class _ObjectClassificationFromSegmentationPipelineImpl(
+    _ObjectClassificationPipelineBase, ObjectClassificationFromSegmentationPipeline
```

Comment:
Not sure if I follow why we need to have two versions of `ObjectClassificationFromSegmentationPipeline`, one public and one private. I'm guessing it's to hide the private methods and potentially also the `__init__`? If so, the private methods are private anyway (or just as private as the private versions of these classes), and the `__init__` seems to be good enough to be public too.